packages/features/data-table/components/DataTableToolbar.tsx

```diff
@@ -39,6 +39,10 @@ function SearchBarComponent({ className }: SearchBarProps, ref: Ref<HTMLInputEle
   const { t } = useLocale();
   const [localValue, setLocalValue] = useState(searchTerm);
 
+  useEffect(() => {
```

Comment:
When filter segments are loaded, it needs to set the search term into this <input> element. So we need to sync it like this.